src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -436,23 +511,29 @@
           (>!! command-send-ch :stop))
         (close! command-send-ch))
 
-      (let [run-interval-minutes (time/in-minutes run-interval)
-            partitioned-hosts (->> hosts
-                                   (map #(assoc % :lastrun (rand-lastrun run-interval)))
-                                   (partition-into-buckets simulation-threads))
-            close-to-stop-chans (mapv (fn [hosts-for-partition]
-                                        (run-simulated-hosts hosts-for-partition
-                                                             run-interval-minutes
-                                                             rand-perc
-                                                             command-send-ch))
-                                      partitioned-hosts)
-            close-to-stop-ch (chan)]
-        (go
-          (<! close-to-stop-ch)
-          (close! rate-monitor-ch)
-          (close! command-send-ch)
-          (dorun (map close! close-to-stop-chans)))
-
+      (let [close-to-stop-ch (chan)
+            [mq-in mq-out mq-errors] (amq-broker-chans "benchmark-broker" "benchmark-endpoint")]
+        ;; initial queue population
+        (println "Initial queue population...")
+        (->> (range numhosts)
```

Comment:
I'd turn this into a `dotimes`, it would get rid of some of the anonymous functions